entrypoint.sh

```diff
@@ -3,10 +3,18 @@ set -euo pipefail
 
 STORAGE_SERVICE_ALIAS=${STORAGE_SERVICE_ALIAS:="s3"}
 STORAGE_SERVICE_URL=${STORAGE_SERVICE_URL:="https://s3.amazonaws.com"}
+SESSION_TOKEN=${SESSION_TOKEN:=""}
 MIRROR_SOURCE=${MIRROR_SOURCE:="."}
+REGION=${REGION:=""}
 
 # Set mc configuration
-mc alias set "$STORAGE_SERVICE_ALIAS" "$STORAGE_SERVICE_URL" "$ACCESS_KEY_ID" "$SECRET_ACCESS_KEY"
+if [ -z "$SESSION_TOKEN" ]; then
+  mc alias set "$STORAGE_SERVICE_ALIAS" "$STORAGE_SERVICE_URL" "$ACCESS_KEY_ID" "$SECRET_ACCESS_KEY"
+  # Execute mc mirror
+  mc mirror $* "$MIRROR_SOURCE" "$STORAGE_SERVICE_ALIAS/$MIRROR_TARGET"
+else
+  export MC_HOST_${STORAGE_SERVICE_ALIAS}=https://${ACCESS_KEY_ID}:${SECRET_ACCESS_KEY}:${SESSION_TOKEN}@s3.${REGION}.amazonaws.com
```

Comment:
This hard codes the `SESSION_TOKEN` functionality to AWS S3 only. It looks like MinIO server [also supports session tokens](https://github.com/minio/mc/blob/master/docs/minio-client-complete-guide.md#specify-temporary-host-configuration-through-environment-variable). Ideally it would be nice to support any S3 compatible backend.

I think there are two options here:
1. Support any S3 backend that has session token functionality.
2. Only support AWS S3 and make that clear in the action interface.

For option 1, we should use `STORAGE_SERVICE_URL` instead of hard coding to `@s3.${REGION}.amazonaws.com`. i.e. Parse the URL and split at `https://`. Add the key, secret, session and `@` symbol.

For option 2, I think we should change the name of `SESSION_TOKEN` and `REGION` to `AWS_SESSION_TOKEN` and `AWS_REGION` to make it clear they only work for AWS S3.

I'm happy with either option. We can always improve it to cover option 1 in the future if you don't want to try and make it work now.

Reply:
I'm happy with option 2 for now, at least we make it visible that it's AWS exclusive.
Option 1 would require a little bit more code and testing, but happy to do it when I get a bit more time.